capstone-app/src/app/modules/maps/map/map.component.ts

```diff
@@ -27,47 +42,122 @@ export class MapComponent {
     setTimeout(() => {
       this.locationSearch();
     });
+    this.nearSearch = new google.maps.places.PlacesService(this.mapComponent._googleMap);
   }
 
   locationSearch() {
     const input = this.searchBar.nativeElement;
+    this.cityLocation = this.searchBar.nativeElement.value;
     const autoComplete = new google.maps.places.Autocomplete(input,
-      {fields: ['geometry', 'name'], types: ['(cities)']});
+      {fields: ['geometry', 'name', 'formatted_address'], types: ['(cities)']});
     autoComplete.addListener('place_changed', () => {
       this.location = autoComplete.getPlace().geometry.location;
+      this.cityLocation = autoComplete.getPlace().formatted_address;
+      this.placesRequestFunc(this.location);
     });
   }
 
+  deleteAllMarks() {
+    this.searchMarkers = [];
+  }
+
+  placesRequestFunc(location: google.maps.LatLng) {
+    this.placesRequest.location = this.location;
+    this.nearSearch.nearbySearch(this.placesRequest, results => {
+      // resetting data
+      this.deleteAllMarks();
+      this.markerData.clear();
+      this.locationNameArray = [];
+      for (const result of results) {
+        this.searchMarkers.push(this.createMarker(result));
+        this.locationNameArray.push(result.name);
+        this.markerData.set(result.name, result.types);
+      }
+    });
+  }
+
+  createMarker(result: google.maps.places.PlaceResult) {
+   return {
+    position: result.geometry.location,
+    title: result.name,
+    icon: {
+        url: "http://maps.google.com/mapfiles/ms/icons/pink-dot.png"
+    }
+   };
+  }
+
+  openInfoWindow(marker: MapMarker) {
+    this.activeMark = marker._marker.getTitle();
+    this.infoWindow.open(marker);
+  }
+
   getCurrentOrSetLocation() {
     if (!this.location) {
-      return new google.maps.LatLng(this.position);
+      if (this.position) {
+        this.location = new google.maps.LatLng(this.position);
+      }
+      else {
+        // default location Bogota, Colombia
+        this.location = new google.maps.LatLng(4.7110, -74.0721);
+      }
     }
-    return this.location;
   }
 
   getCurrentLocation () {
     navigator.geolocation.getCurrentPosition(position => {
-    this.position = {
+      this.position = {
         lat: position.coords.latitude,
         lng: position.coords.longitude,
-      }
-    })
+      };
+      this.location = new google.maps.LatLng(this.position);
+      this.getCurrentOrSetLocation();
+      this.geocoder.geocode({'location': this.position}, function(results, status) {
+        if (status === 'OK') {
+          this.cityLocation = results[5].formatted_address;
```

Comment:
Like we discussed offline, this doesn't seem very resilient to me. Consider replacing this with a filter that removes results with numbers in it (to eliminate addresses and zip codes), and takes the first option that does not have those.